src/Sulu/Bundle/SecurityBundle/AccessControl/AccessControlQueryEnhancer.php

```diff
@@ -115,12 +115,12 @@ private function enhanceQueryWithAccessControl(
             'WITH',
             $accessClassCondition . ' AND ' . $this->getEntityIdCondition($entityClass, 'entity', $entityIdField)
         );
-        $subQueryBuilder->leftJoin('accessControl.role', 'role', 'WITH', 'role.system = :system');
+        $subQueryBuilder->leftJoin('accessControl.role', 'role');
```

Comment:
with the previous statement, if an access control entity existed in another system, the `leftJoin` joins an empty row here. the empty row will be interpreted as "entity cannot be accessed" by the `OR role.id IS NULL` below. 

this logic was most likely a leftover from the previous implementation that was changed in https://github.com/sulu/sulu/pull/6364

Reply:
I'm missing something or can now the 2 joins be `innerJoin` instead of `leftJoin`?

Reply:
yes you are right, that should be a bit more performant 🙂